apis/refs/v1beta1/computerefs.go

```diff
@@ -27,6 +27,15 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// fixStaleExternalFormat converts the "External" reference field to the right format if a SelfLink value is used.
+// This guarantees the backward compatibility for Compute Beta resources.
+func fixStaleExternalFormat(external string) string {
+	external = strings.TrimPrefix(external, "https://www.googleapis.com/compute/v1/")
+	external = strings.TrimPrefix(external, "https://www.googleapis.com/compute/v1beta1/")
+	external = strings.TrimPrefix(external, "/")
```

Comment:
allow user to specify something like `/projects/<projectID>/global/networks/<networkID>'. This is trivial but with a nicer experience.

Reply:
I'm unsure if we want to fix in this way. As discussed before, we do want to have strict format restriction on `external` field, at least for new resources.  I'm afraid allowing stale formats would break this expectation? 

I feel we may want to have a special handling for backward capability(existing Beta resources), write a separate network resolver accepts other formats, like what I had for forwarding rule([code](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/5b1307cd3451ec9fe3c252707f2a2ff32deaa3d9/pkg/controller/direct/compute/refs.go)) or reuse the `ResolveComputeNetwork` in SQLInstance.

Reply:
Are you talking about distinguishing between existing object and new object? Yes, I like that idea. Though this would require each resource that references ComputeNetwork to change the ResolveRef between "create" and other functions. I don't know how feasible that would be. 
 
Another question is how much do we know about the abnormal format of the Compute resource "External" field.

Reply:
I do not have a 100% sure answer for the first part, but since we just started the Beta->direct migration for a few resources, if we have a standard to follow now, then there will not be too much work.

As far as I know, allowing "selfLink" as External is very common for Compute resources and also for other resources.